File: app/src/main/java/savindev/myuniversity/serverTasks/ScheduleDates.java
package savindev.myuniversity.serverTasks;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;

public class ScheduleDates { //Класс для парсинга дополнительных расписаний, по датам

    public final int ID_SCHEDULE;
    public final ArrayList<DATE> DATES;

    public ScheduleDates(int ID_SCHEDULE, JSONArray dates) {
        this.ID_SCHEDULE = ID_SCHEDULE;
        this.DATES = DATE.fromJson(dates);
    }

    public static ScheduleDates fromJson(final JSONObject object) {
        final int ID_SCHEDULE = object.optInt("ID_SCHEDULE", 0);
        JSONArray DATES = null;
        try {
            DATES = object.getJSONArray("DATES");
        } catch (JSONException e) {
            e.printStackTrace();
        }
        return new ScheduleDates(ID_SCHEDULE, DATES);
    }

    public static ArrayList<ScheduleDates> fromJson(final JSONArray array) {
        final ArrayList<ScheduleDates> ScheduleDates = new ArrayList<>();
        for (int index = 0; index < array.length(); ++index) {
            try {
                final ScheduleDates sched = fromJson(array.getJSONObject(index));
                if (null != sched) ScheduleDates.add(sched);
            } catch (final JSONException ignored) {
            }
        }
        return ScheduleDates;
    }

    public static class DATE { //Класс для парсинга дополнительных расписаний, по датам

        public final String DATE;
        public final boolean IS_CANCELED;
        public final boolean IS_DELETED;

        public DATE(String DATE, boolean IS_CANCELED, boolean IS_DELETED) {
            this.DATE = DATE;
            this.IS_CANCELED = IS_CANCELED;
            this.IS_DELETED = IS_DELETED;
        }

        public static DATE fromJson(final JSONObject object) {
            final String DATE = object.optString("DATE", "");
            final boolean IS_CANCELED = object.optBoolean("IS_CANCELED", false);
            final boolean IS_DELETED = object.optBoolean("IS_DELETED", false);
            return new DATE(DATE, IS_CANCELED, IS_DELETED);
        }

        public static ArrayList<DATE> fromJson(final JSONArray array) {
            final ArrayList<DATE> dates = new ArrayList<>();
            for (int index = 0; index < array.length(); ++index) {
                try {
                    final DATE data = fromJson(array.getJSONObject(index));
                    if (null != data) dates.add(data);
                } catch (final JSONException ignored) {
                }
            }
            return dates;
        }
    }
}

File: app/src/main/java/savindev/myuniversity/serverTasks/ScheduleDates.kt
package savindev.myuniversity.serverTasks

import org.json.JSONArray
import org.json.JSONException
import org.json.JSONObject
import java.util.*

class ScheduleDates(//Класс для парсинга дополнительных расписаний, по датам

        val ID_SCHEDULE: Int, dates: JSONArray) {
    val DATES: ArrayList<DATE>

    init {
        this.DATES = DATE.fromJson(dates)
    }

    class DATE(//Класс для парсинга дополнительных расписаний, по датам

            val DATE: String, val IS_CANCELED: Boolean, val IS_DELETED: Boolean) {
        companion object {

            fun fromJson(`object`: JSONObject): DATE? {
                val DATE = `object`.optString("DATE", "")
                val IS_CANCELED = `object`.optBoolean("IS_CANCELED", false)
                val IS_DELETED = `object`.optBoolean("IS_DELETED", false)
                return DATE(DATE, IS_CANCELED, IS_DELETED)
            }

            fun fromJson(array: JSONArray): ArrayList<DATE> {
                val dates = ArrayList<DATE>()
                for (index in 0..array.length() - 1) {
                    try {
                        val data = fromJson(array.getJSONObject(index))
                        if (null != data) dates.add(data)
                    } catch (ignored: JSONException) {
                    }

                }
                return dates
            }
        }
    }

    companion object {

        fun fromJson(`object`: JSONObject): ScheduleDates? {
            val ID_SCHEDULE = `object`.optInt("ID_SCHEDULE", 0)
            var DATES: JSONArray? = null
            try {
                DATES = `object`.getJSONArray("DATES")
            } catch (e: JSONException) {
                e.printStackTrace()
            }

            return ScheduleDates(ID_SCHEDULE, DATES!!)
        }

        fun fromJson(array: JSONArray): ArrayList<ScheduleDates> {
            val ScheduleDates = ArrayList<ScheduleDates>()
            for (index in 0..array.length() - 1) {
                try {
                    val sched = fromJson(array.getJSONObject(index))
                    if (null != sched) ScheduleDates.add(sched)
                } catch (ignored: JSONException) {
                }

            }
            return ScheduleDates
        }
    }
}